node/node_test.go

```diff
@@ -4048,6 +4051,308 @@ func TestNode_setTxGuardianData(t *testing.T) {
+			node.WithBootstrapComponents(bootstrapComponents),
+		)
+		guardianData, blockInfo, err := n.GetGuardianData(userAddress, api.AccountQueryOptions{})
+		require.Equal(t, api.GuardianData{
+			ActiveGuardian:  nil,
+			PendingGuardian: apiG1,
+			Frozen:          false,
+		}, guardianData)
+		require.Equal(t, api.BlockInfo{}, blockInfo)
+		require.Nil(t, err)
+	})
+	t.Run("one active and one pending", func(t *testing.T) {
+		bootstrapComponents := getDefaultBootstrapComponents()
+		bootstrapComponents.GuardedAccountHandlerField = &guardianMocks.GuardedAccountHandlerStub{
+			GetConfiguredGuardiansCalled: func(uah state.UserAccountHandler) (active *guardians.Guardian, pending *guardians.Guardian, err error) {
+				return g1, g2, nil
+			},
+		}
+		n, _ := node.NewNode(
+			node.WithDataComponents(dataComponents),
+			node.WithCoreComponents(coreComponents),
+			node.WithStateComponents(stateComponents),
+			node.WithBootstrapComponents(bootstrapComponents),
+		)
+		guardianData, blockInfo, err := n.GetGuardianData(userAddress, api.AccountQueryOptions{})
+		require.Equal(t, api.GuardianData{
+			ActiveGuardian:  apiG1,
+			PendingGuardian: apiG2,
+			Frozen:          false,
+		}, guardianData)
+		require.Equal(t, api.BlockInfo{}, blockInfo)
+		require.Nil(t, err)
+	})
+	t.Run("one active and one pending and account frozen", func(t *testing.T) {
+		acc, _ := state.NewUserAccount(userAddressBytes)
+		acc.CodeMetadata = (&vmcommon.CodeMetadata{Frozen: true}).ToBytes()
+		accDB := &stateMock.AccountsStub{
+			GetAccountWithBlockInfoCalled: func(address []byte, options common.RootHashHolder) (vmcommon.AccountHandler, common.BlockInfo, error) {
+				return acc, nil, nil
+			},
+			RecreateTrieCalled: func(_ []byte) error {
+				return nil
+			},
+		}
+		stateComponents := getDefaultStateComponents()
+		args := state.ArgsAccountsRepository{
+			FinalStateAccountsWrapper:      accDB,
+			CurrentStateAccountsWrapper:    accDB,
+			HistoricalStateAccountsWrapper: accDB,
+		}
+		stateComponents.AccountsRepo, _ = state.NewAccountsRepository(args)
+		bootstrapComponents := getDefaultBootstrapComponents()
+		bootstrapComponents.GuardedAccountHandlerField = &guardianMocks.GuardedAccountHandlerStub{
+			GetConfiguredGuardiansCalled: func(uah state.UserAccountHandler) (active *guardians.Guardian, pending *guardians.Guardian, err error) {
+				return g1, g2, nil
+			},
+		}
+		n, _ := node.NewNode(
+			node.WithDataComponents(dataComponents),
+			node.WithCoreComponents(coreComponents),
+			node.WithStateComponents(stateComponents),
+			node.WithBootstrapComponents(bootstrapComponents),
+		)
+		guardianData, blockInfo, err := n.GetGuardianData(userAddress, api.AccountQueryOptions{})
+		require.Equal(t, api.GuardianData{
+			ActiveGuardian:  apiG1,
+			PendingGuardian: apiG2,
+			Frozen:          true,
+		}, guardianData)
+		require.Equal(t, api.BlockInfo{}, blockInfo)
+		require.Nil(t, err)
+	})
+}
+
+func TestNode_getPendingAndActiveGuardians(t *testing.T) {
```

Comment:
kind of duplicated tests. Can we keep only the first set of tests? TestNode_GetGuardianData